internal/controller/nodeagent.go

```diff
@@ -270,22 +286,57 @@ func (r *DataProtectionApplicationReconciler) customizeNodeAgentDaemonset(ds *ap
 		},
 	)
 
+	privileged := true
+
+	if dpa.Spec.Configuration.Velero.DisableFsBackup != nil {
+		privileged = !*dpa.Spec.Configuration.Velero.DisableFsBackup
+	}
+
+	ds.Spec.Template.Spec.SecurityContext = &corev1.PodSecurityContext{
+		RunAsNonRoot: ptr.To(!privileged),
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeRuntimeDefault,
+		},
+		SupplementalGroups: dpa.Spec.Configuration.NodeAgent.SupplementalGroups,
+	}
+
+	if privileged {
+		ds.Spec.Template.Spec.SecurityContext.RunAsUser = ptr.To(int64(0))
+		// Privileged containers always run as Unconfined seccomp profile
+		// Changing to match the default behavior of the privileged node-agent
+		ds.Spec.Template.Spec.SecurityContext.SeccompProfile = &corev1.SeccompProfile{
+			Type: corev1.SeccompProfileTypeUnconfined,
```

Comment:
Earlier we were not setting this explicitly, right ?

Reply:
No we were not. We could omit that part, and it would still become Unconfined by default. However, I believe it's better to maintain consistent behavior - if we set SSC for the non-privileged case, we should do the same for this use case as well:

https://kubernetes.io/docs/tutorials/security/seccomp/#before-you-begin

```
Note:
It is not possible to apply a seccomp profile to a container running
with privileged: true set in the container's securityContext.
Privileged containers always run as Unconfined.
```